Mathlib/Algebra/Polynomial/Lifts.lean

```diff
@@ -228,4 +228,25 @@ theorem smul_mem_lifts {p : S[X]} (r : R) (hp : p ∈ lifts (algebraMap R S)) :
 
 end Algebra
 
+section IsSimpleRing
+
+variable {R S : Type*} [CommRing R] [IsSimpleRing R] [CommRing S] [Nontrivial S] [Algebra R S]
+
+theorem monic_of_eq_finprod (p : R[X]) {n : ℕ} (b : Fin n → S)
+    (hp : mapAlg R S p = finprod fun k : Fin n ↦ X - C (b k)) : p.Monic := by
```

Comment:
I think you could add `Finset.prod` and `finprod` versions of [Polynomial.monic_prod_multiset_X_sub_C](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Polynomial/Roots.html#Polynomial.monic_prod_multiset_X_sub_C), and separately add an `mapAlg` version of `monic_of_injective` (that assumes [Semiring R] [Semiring S] [FaithfulSMul R S]), and then this theorem and the next could be removed, because the proofs would be just something like
```
`monic_of_faithfulSMul (hp ▸ monic_finprod_multiset_X_sub_C _ _ fun m _ ↦ monic_X_sub_C (b m))`